backend/src/stwo/prover.rs

```diff
@@ -317,36 +388,63 @@ where
         );
 
         let config = get_config();
-        let proof: StarkProof<MC::H> =
+
+        let proof: Proof<MC> =
             bincode::deserialize(proof).map_err(|e| format!("Failed to deserialize proof: {e}"))?;
 
         let verifier_channel = &mut <MC as MerkleChannel>::C::default();
         let commitment_scheme = &mut CommitmentSchemeVerifier::<MC>::new(config);
 
         //Constraints that are to be proved
-        let component = PowdrComponent::new(
-            &mut TraceLocationAllocator::default(),
-            PowdrEval::new(self.analyzed.clone()),
-            (SecureField::zero(), None),
-        );
 
-        // Retrieve the expected column sizes in each commitment interaction, from the AIR.
-        // the sizes include the degrees of the constant, witness, native lookups. Native lookups are not used yet.
-        let sizes = component.trace_log_degree_bounds();
+        let tree_span_provider = &mut TraceLocationAllocator::default();
+        let mut components = Vec::new();
+
+        let mut constant_cols_offset_acc = 0;
+        let mut index_machine = 0;
+
+        self.split.iter().for_each(|(_, pil)| {
```

Comment:
This could be a map instead of a `for_each`, returning `components` instead of using side effects.

```
let components = self.split.iter().map(|...| ...).collect();
```

Reply:
Also, instead of incrementing `index_machines` by hand, we could use `self.split.iter().zip_eq(proof.sizes[MACHINE_SIZES_IDS])`